main/CWcomparison.m

```diff
@@ -36,17 +36,26 @@
 CWsolution = zeros(size(hillVecs));
 curviCWsolution = CWsolution;
 CW2solution = zeros(3,length(ref_t));
+CWsolution2 = zeros(size(hillVecs));
 for j = 1:length(ref_t)
  t = ref_t(1) - ref_t(j);
  hill_matrix = make_hill_matrix(n, t);
  CWsolution(:, j) = hill_matrix * hillVecs(:, j);
  curviCWsolution(:, j) = hill_matrix * curvHillVecs(:, j);
  CW2solution(:,j) = cw_2nd_order(hillVecs(:,j),n,rad+1200,t);
+ if j>1 && j<length(ref_t)-1

```

Comment:
This loop only propagates a single solution (the worst one I believe) and doesn't use the small step hill matrix.